sdk/core/src/main/java/com/klaviyo/coresdk/lifecycle/KlaviyoLifecycleMonitor.kt

```diff
@@ -0,0 +1,44 @@
+package com.klaviyo.coresdk.lifecycle
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+
+/**
+ * Service for monitoring the application lifecycle and network connectivity
+ */
+internal object KlaviyoLifecycleMonitor : LifecycleMonitor, Application.ActivityLifecycleCallbacks {
+
+    private var activitiesActive = 0
+
+    private var whenStopped = mutableListOf<ActivityObserver>()
+
+    override fun whenStopped(observer: ActivityObserver) {
+        whenStopped += observer
+    }
```

Comment:
should there be a way to remove an observer? when will this list get cleared?

Reply:
Yes... I skipped that because the observers we need will live for the life of the application, but yes I should at least write the remove method.

Reply:
Come to think of it, maybe the listeners should be removed or paused when the activity sleeps... I'll need to read up on lifecycle a little more.

Reply:
Added removal option.